src/core/public/http/fetch.ts

```diff
@@ -148,10 +149,19 @@ export class Fetch {
       }),
     };
 
-    const url = format({
-      pathname: shouldPrependBasePath ? this.params.basePath.prepend(options.path) : options.path,
-      query: removedUndefined(query),
-    });
+    let url;
+    if (options.headers?.withoutClientBasePath) {
+      const theoptions: PrependOptions = { withoutClientBasePath: true };
+      url = format({
+        pathname: this.params.basePath.remove(options.path, theoptions),
+        query: removedUndefined(query),
+      });
+    } else {
+      url = format({
+        pathname: shouldPrependBasePath ? this.params.basePath.prepend(options.path) : options.path,
+        query: removedUndefined(query),
+      });
+    }
```

Comment:
```suggestion
    const prependOptions = options.withoutClientBasePath ? { withoutClientBasePath: options.withoutClientBasePath } : undefined;
    const url = format({
        pathname: shouldPrependBasePath ? this.params.basePath.prepend(options.path, prependOptions) : options.path,
        query: removedUndefined(query),
      });
```
clientBasePath is part of basePath, basePath = `{serverBasePath}{clientBasePath}`.